app/scripts/lib/setupSentry.js

```diff
@@ -127,6 +127,71 @@ export const SENTRY_UI_STATE = {
   unconnectedAccount: true,
 };
 
+/**
+ * Returns whether MetaMetrics is enabled, given the application state.
+ *
+ * @param {{ state: unknown} | { persistedState: unknown }} appState - Application state
+ * @returns `true` if MetaMask's state has been initialized, and MetaMetrics
+ * is enabled, `false` otherwise.
+ */
+function getMetaMetricsEnabledFromAppState(appState) {
+  // during initialization after loading persisted state
+  if (appState.persistedState) {
+    return getMetaMetricsEnabledFromPersistedState(appState.persistedState);
+    // After initialization
+  } else if (appState.state) {
+    // UI
+    if (appState.state.metamask) {
+      return Boolean(appState.state.metamask?.participateInMetaMetrics);
```

Comment:
Unnecessary `?`, we checked `appState.state.metamask` already in the condition
```suggestion
      return Boolean(appState.state.metamask.participateInMetaMetrics);
```